apm-agent-core/src/main/java/co/elastic/apm/agent/impl/error/ErrorCapture.java

```diff
@@ -44,6 +44,8 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ErrorCapture.class);
 
+    private static final ThreadLocal<ErrorCapture> activeError = new ThreadLocal<>();
```

Comment:
You reminded me that this one made me think I need to take another look into our thread local usages in general. 
In this specific case, I don't think it is required, as it is set and removed atomically within a single method execution. Once a method bytecode starts to execute, I don't think any of its dependencies may change before it ends.
But it may be something we want to set general rules for and make sure to follow all across.